File: backend/src/test/java/com/kiwi/users/UsersTestFactory.java
package com.kiwi.users;

import com.kiwi.features.users.data.LoginDTO;
import com.kiwi.features.users.data.UsersDTO;

import java.time.LocalDate;

import static com.kiwi.common.utils.FormatUtils.formatDate;

public class UsersTestFactory {
    public static LoginDTO validLoginDTO() {
        return new LoginDTO("finn@thehuman.com", "Math3matic!");
    }

    public static LoginDTO invalidLoginDTO() {
        return new LoginDTO("football", "kk");
    }

    private static final String DATE = formatDate(LocalDate.of(2025, 1, 1));

    public static UsersDTO validUserDTO() {
        return new UsersDTO(validLoginDTO().getEmail(), DATE);
    }

    public static UsersDTO invalidUserDTO() {
        return new UsersDTO(invalidLoginDTO().getEmail(), DATE);
    }
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/users/tests/UsersTestFactory.kt
package com.bellako.kiwi.features.users.tests

import com.bellako.kiwi.features.users.data.LoggedDTO
import com.bellako.kiwi.features.users.data.LoginDTO
import com.bellako.kiwi.features.users.data.UsersDTO

object UsersTestFactory {
    fun validLoginDTO(): LoginDTO =
        LoginDTO(
            email = "finn@thehuman.com",
            password = "Math3matic!",
        )

    fun invalidLoginDTO(): LoginDTO =
        LoginDTO(
            email = "football.com",
            password = "kk",
        )

    private const val DATE = "2025-01-01"

    fun validLoggedDTO(): LoggedDTO =
        LoggedDTO(
            jwt = "test",
            registerDate = DATE,
        )

    fun validUsersDTO(): UsersDTO =
        UsersDTO(
            email = validLoginDTO().email,
            password = validLoginDTO().password,
            registerDate = DATE,
        )

    fun invalidUsersDTO(): UsersDTO =
        UsersDTO(
            email = invalidLoginDTO().email,
            password = invalidLoginDTO().password,
            registerDate = DATE,
        )
}
